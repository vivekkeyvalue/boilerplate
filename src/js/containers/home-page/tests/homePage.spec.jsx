import React from 'react';
import { shallow, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import HomePage from '../homePage';
import { Button } from '../../../components';

configure({ adapter: new Adapter() });

const mockClick = jest.fn();


describe('components/homePage', () => {
  it('Renders default options', () => {
    const component = shallow(
      <HomePage />
    );
    const tree = shallowToJson(component);
    expect(tree).toMatchSnapshot();
  });
  it('Renders with click function and message', () => {
    const dummyData = {
      message: 'hello world!'
    };
    const component = shallow(
      <HomePage fetchDummyApi={mockClick} dummyData={dummyData} />
    );
    component.find(Button).props().onClick();
    expect(mockClick).toHaveBeenCalled();
  });
});
