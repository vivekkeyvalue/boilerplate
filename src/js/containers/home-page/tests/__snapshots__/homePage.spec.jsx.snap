// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/homePage Renders default options 1`] = `
<div
  className="container"
>
  <div
    className="wrapper"
  >
    <div
      className="header"
    >
      <img
        alt="logo"
        className="logo"
        src="assets/logo.svg"
      />
      <div>
         Welcome to React 
      </div>
    </div>
    <p
      className="intro"
    >
      Start building with new &lt;React /&gt;  and &lt;Boilerplate /&gt;
    </p>
    <Button
      label="Click Me!"
      onClick={[Function]}
    />
  </div>
</div>
`;
